### ISSUE TITLE
Autocomplete two closing brackets

### ISSUE THREAD

**Issue by entr0pia** [VSCodeVim/Vim#8748]
**Describe the bug**
With **Microsoft Pinyin** input method on Window 10 (or use **Old version** on Windows 11), when typing an opening bracket `(`, two closing brackets `))` are automatically completed. Square brackets, curly brackets, and sometimes quotation marks also have the same bug.

**To Reproduce**

1. With **Microsoft Pinyin** input method on Window 10 (or use **Old version** on Windows 11)
2. Use **English punctuation** when inputting Chinese
3. Switch to Chinese mode
4. Type an opening bracket at INSERT mode of vim
5.  See error

**Expected behavior**
Automatically completed **only one** closing brackets

**Screenshots**
![bug](https://github.com/VSCodeVim/Vim/assets/30486766/1bff46cf-9724-4cf3-983a-fec3c8f1ba0e)

**Environment (please complete the following information):**

- Extension (VsCodeVim) version: v1.6.2 to v1.27.1, at least
- VSCode version: 1.84.2 (user setup)
- OS: Windows 11 23H2, Windows 10 21H2

**Additional context**
I tested and reproduced it with only the vim plugin enabled.

**Comment by cyan0714** [comment]
I have the same question, did you solve it ?

**Comment by lhs-12** [comment]
Same problem. I'm using the RIME input method, even when the input mode is set to English, this problem will still be triggered.

**Comment by Takohubuki** [comment]
> Same problem. I'm using the RIME input method, even when the input mode is set to English, this problem will still be triggered.

same problem, guess its about RIME. 'cause it won't happen when using microsoft IME.

**Comment by lhs-12** [comment]
> > Same problem. I'm using the RIME input method, even when the input mode is set to English, this problem will still be triggered.
> 
> same problem, guess its about RIME. 'cause it won't happen when using microsoft IME.

I don't think so. I still use Rime input method and didn't change the configuration, but the problem disappeared one day, and my VsCode Vim doesn't have this problem now.
